File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/InboxConfig.cs
using System;
using JetBrains.Annotations;

namespace Volo.Abp.EventBus.Distributed;

public class InboxConfig
{
    [NotNull]
    public String Name;//  { get; }

    public Type ImplementationType;// { get; set; }

    public Func<Type, bool> EventSelector;// { get; set; }

    public Func<Type, bool> HandlerSelector;// { get; set; }

    /**
     * Used to enable/disable processing incoming events.
     * Default: true.
    */
    public boolean IsProcessingEnabled;// { get; set; } = true;

    public InboxConfig(@Nonnull String name)
    {
        Name = Check.NotNullOrWhiteSpace(name, nameof(name));
    }
}

File: ueueo-framework/ueueo-eventbus/src/main/java/com/ueueo/eventbus/distributed/InboxConfig.java
package com.ueueo.eventbus.distributed;

import lombok.Data;
import org.springframework.lang.NonNull;

import java.util.Objects;
import java.util.function.Function;

@Data
public class InboxConfig {
    @NonNull
    private String name;

    private Class<?> implementationType;

    private Function<Class<?>, Boolean> eventSelector;

    private Function<Class<?>, Boolean> handlerSelector;

    /**
     * Used to enable/disable processing incoming events.
     * Default: true.
     */
    private boolean isProcessingEnabled = true;

    public InboxConfig(@NonNull String name) {
        Objects.requireNonNull(name);
        this.name = name;
    }
}
